tf2/src/buffer_core.cpp

```diff
@@ -231,36 +231,36 @@ bool BufferCore::setTransformImpl(
     error_exists = true;
   }
 
-  if (std::isnan(transform_in.getOrigin().x()) || std::isnan(transform_in.getOrigin().y()) ||
-    std::isnan(transform_in.getOrigin().z()) ||
-    std::isnan(transform_in.getRotation().x()) || std::isnan(transform_in.getRotation().y()) ||
-    std::isnan(transform_in.getRotation().z()) || std::isnan(transform_in.getRotation().w()))
+  if (std::isnan(origin_in.x()) || std::isnan(origin_in.y()) ||
+    std::isnan(origin_in.z()) ||
+    std::isnan(rotation_in.x()) || std::isnan(rotation_in.y()) ||
+    std::isnan(rotation_in.z()) || std::isnan(rotation_in.w()))
   {
     RCUTILS_LOG_ERROR(
       "TF_NAN_INPUT: Ignoring transform for child_frame_id \"%s\" from authority \"%s\" because"
       " of a nan value in the transform (%f %f %f) (%f %f %f %f)",
       stripped_child_frame_id.c_str(), authority.c_str(),
-      transform_in.getOrigin().x(), transform_in.getOrigin().y(), transform_in.getOrigin().z(),
-      transform_in.getRotation().x(), transform_in.getRotation().y(),
-      transform_in.getRotation().z(), transform_in.getRotation().w()
+      origin_in.x(), origin_in.y(), origin_in.z(),
```

Comment:
`isNan()` or `isValid()` on Vector/Quaternion would be really nice for this, along with a normalization check on quaternion - this code doesn't belong here.

Reply:
Can you file an issue and use the `good first issue` tag? I can throw it to a student who wants to get their feet wet.

Reply:
Filed, I don't have permission to add tags, though

Reply:
https://github.com/ros2/geometry2/issues/742

Reply:
Done. Thanks.